serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc

```diff
@@ -0,0 +1,359 @@
+= Timeouts on events for {context}
+:compat-mode!:
+// Metadata:
+:description: Using timeouts in {context}
+:keywords: kogito, workflow, serverless, timeout, timer, expiration
+
+When defining a state in a workflow you can set a timeout which is a property used to control the maximum time this state should take to be complete. In case of the time is elapsed and overdue, this state is considered timed-out, and the workflow moves to the next stated defined in the transition. The properties that you can use when defining a state timeout are described in the link:{spec_doc_url}#event-timeout-definition[Serverless Workflow specification].
+
+The event-based states can define a property `eventTimeout` which uses ISO 8601 data and time format to specify a duration of time, for instance, you can if you can represent 15 minutes like `PT15M`,  to represent 15 minutes or 2 days, 3 hours and 4 minutes `P2DT3H4M`.
+
+[NOTE]
+====
+Event timeouts can not be defined as a specific point in time, but it should be an amount of time, a duration, which is considered to start when the referred state becomes active in the workflow.
+====
+
+[IMPORTANT]
+====
+{product_name} currently, supports timeout only for *Callback* and *Switch* states with events, other states are going to support timeouts in the future releases.
+====
+
+=== Callback state timeout
+Callback state can be used when you need to publish an event to trigger an external service and waits an asynchronous response as an event, the callback. Once this event is consumed, the workflow continues the execution to the next state defined in the transition property. See more on xref:eventing/working-with-callbacks.adoc[Callback state in {context}].
```

Comment:
Maybe

"Callback state can be used when you need to execute an action, for example invoke an external service, and wait for an asynchronous response in the form of an event, the callback"

Reply:
Maybe
"When the event arrives, it is consumed, and the workflow continues...."

Reply:
thanks